requirements/0004-volunteer--host-identity-validated.md

```diff
@@ -0,0 +1,175 @@
+
+
+##### ... I want to recover my account
+
+When user tries to register with existing phone number system SHALL show Account Recovery View.
+Account Recovery View SHALL notify user that this option is not available and to call Crisis Point's Hot Line.
+
+### User Interface
+
+No impact.
+
+#### API
+
+New endpoints SHALL be created for this feature. These endpoints SHALL be available without authentication.
+
+### Services
+
+#### New Host Registration Service (POST /registration)
+
+If phone number sent in POST data already exists in database system SHALL return HTTP 302 and re-render generic Account Recovery View.
+
+If phone number sent in POST data does not exist in database, System:
+- SHALL create *Host* entry using data from POST, and set its status to Verification Pending.
+- SHALL create *AccommodationUnit* entry using data from POST, and set its status to Verification Pending.
+- SHALL create *VerificationSession* entry referencing newly created *Host* and *AccommodationUnit*.
+- SHALL use newly created *VerificationSession*'s GUID to form `callback` pointing at `POST /registration/{GUID}` endpoint.
+- SHALL use newly created *Host*'s GUID to form `returnUrl` pointing at Host Verification Status View.
+
+#### Host Identity Verification Service (POST /registration/{verificationSessionId})
+
+System SHALL only accept valid requests. Sender verification per [Authologic documentation](https://sandbox.authologic.com/docs/api/pl/v1.1/api/callback.html#weryfikacja-nadawcy) SHALL be determined as follows:
+- `X-Signature-Timestamp` SHALL be no more than 5 minutes different from server time when last byte of message was received.
+- `HMAC_SHA_256` hash of `<X-Signature-Timestamp>:<HTTP Response Body>` using secret signature key SHALL be equal to `X-Signature` header. When calculating the hash raw HTTP Body exactly as in Response SHALL be used, without any reformatting that might be introduced by backend frameworks.
+
+When valid response is received from Authologic, system SHALL examine `result.identity.status` field of the response. System SHALL set verification status of both *Host* and *AccommodationUnit* correlated with given *VerificationSession* to:
+- Verified if received status is `FINISHED` 
+- Rejected if received status is `FAILED`
+- Verification pending if received status is `IN_PROGRESS` or `PARTIAL`
+
+When *Host* verification status is changed to Verified, system SHALL send and email to the given email address notifying a *Host* of the fact.
+
+### Availability
+
+No impact.
+
+### Security
+
+Callback endpoint SHALL verify authenticity of the request with HMAC_SHA_256 checksum using a dedicated signature key.
+
+Registration endpoint SHALL always return HTTP 302 status code to protect from brute-force account enumeration attacks.
```

Comment:
Do we care about response timing potentially allowing for enumeration?

Reply:
That might be a good idea. Not sure how to implement it tho. GCP by design provides some DDoS. For functions, where would we do some backoff mechanism?

Reply:
Wait, this whole point contradicts with redirecting to a different view when a host with given email/phone is already registered